File: Leetcode Contest/Biweekly Contest 174/Best Reachable Tower.java
class Solution {
    public int[] bestTower(int[][] towers, int[] center, int radius) {
        int bestQuality = -1;
        int bestX = -1, bestY = -1;

        int cx = center[0];
        int cy = center[1];

        for (int[] tower : towers) {
            int x = tower[0];
            int y = tower[1];
            int q = tower[2];

            int dist = Math.abs(x - cx) + Math.abs(y - cy);

            if (dist <= radius) {
                if (q > bestQuality ||
                   (q == bestQuality && (x < bestX || (x == bestX && y < bestY)))) {
                    bestQuality = q;
                    bestX = x;
                    bestY = y;
                }
            }
        }

        if (bestQuality == -1) {
            return new int[]{-1, -1};
        }

        return new int[]{bestX, bestY};
    }
}
©leetcode

File: Leetcode Contest/Biweekly Contest 174/Best Reachable Tower.py
from typing import List

class Solution:
    def bestTower(self, towers: List[List[int]], center: List[int], radius: int) -> List[int]:
        best_quality = -1
        best_x = -1
        best_y = -1

        cx, cy = center

        for x, y, q in towers:
            dist = abs(x - cx) + abs(y - cy)

            if dist <= radius:
                if (
                    q > best_quality or
                    (q == best_quality and (x < best_x or (x == best_x and y < best_y)))
                ):
                    best_quality = q
                    best_x = x
                    best_y = y

        if best_quality == -1:
            return [-1, -1]

        return [best_x, best_y]
